Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -219,35 +228,21 @@ void uartStyleDecoder::setParityMode(UartParity newParity)
     parity = newParity;
 }
 
-void uartStyleDecoder::performParityCheck()
+bool uartStyleDecoder::isParityCorrect(uint32_t bitField) const
 {
-    auto isEvenParity = [=] () -> bool
-    {
-        uint32_t mask = 0x00000001;
-        uint8_t parity = 0;
-        for (int i = 0; i < dataBit_max; i++)
-        {
-            const uint8_t currentBit = (dataBit_current & mask) ? 1 : 0;
-            parity = parity ^ currentBit;
-            mask = mask << 1;
-        }
-
-        return parity == 0;
-    };
-
-    switch(parity)
-    {
-    case UartParity::None:
-        assert(false);
-        return;
-    case UartParity::Even:
-        parityCheckFailed = ! isEvenParity();
-    case UartParity::Odd:
-        parityCheckFailed = isEvenParity();
-    }
+	if (parity == UartParity::None)
+		std::terminate();
 
-    return;
+	return parityOf(bitField) == parity;
 }
 
+UartParity uartStyleDecoder::parityOf(uint32_t bitField) const
+{
+	bool result = false;
```

Comment:
I'm not sure that `result` is the best name for the bool here.  Maybe something like `isOddParity`?